awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -95,6 +97,9 @@ def __init__(self, argument_spec, **kwargs):
 
         self.session = Request(cookies=CookieJar(), validate_certs=self.verify_ssl)
 
+        # Load the ping page and check the module
+        self.check_version()
```

Comment:
this is going to add 1 request to every module execution. That's a pretty step performance penalty. Makes me want the API itself to return version data in headers.

Reply:
If you are going to change anything with this, also add some indicator if its AWX or "Tower".

Reply:
If we want to do this, we should definitely find a way to do it that doesn't involve 2x request overhead.

Reply:
Calculating this:

> add some indicator if its AWX or "Tower".

...per request is a little tricky because determining if it's AWX or not depends on licensing code that can be expensive to run per-request.

Reply:
https://github.com/ansible/awx/pull/6788